test/system/posts_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'application_system_test_case'
+
+class PostsTest < ApplicationSystemTestCase
+  setup do
+    @userone = users(:one)
+    @usertwo = users(:two)
+    @postone = posts(:pone)
+    # @posttwo = posts(:ptwo)
+  end
+  test 'Post index page' do
+    visit user_posts_path(@userone)
+    assert 'avator.png', @userone.photo
+    assert_selector 'a', text: @userone.name
+    assert_selector 'p', text: "Number of post #{@userone.posts_counter}"
+
+    @userone.posts.each do |post|
+      assert_selector 'a', text: post.title
+      assert_selector 'p', text: post.text
+      assert_selector 'span', text: post.comments.first.text
+      assert_selector 'span', text: post.comments_counter
+      assert_selector 'span', text: post.likes_counter
+    end
+    assert_selector 'a', text: 'Pagination'
+    click_link @userone.posts.first.title, match: :first
+    assert user_post_path(@userone, @userone.posts.first)
+  end
+
+  test 'post show page' do
+    visit user_post_path(@usertwo, @usertwo.posts.first)
+    assert_selector 'strong', text: @usertwo.name
+    assert_selector 'strong', text: @usertwo.posts.first.title
+    assert_selector 'span', text: @usertwo.posts.first.comments_counter
+    assert_selector 'span', text: @usertwo.posts.first.likes_counter
+    assert_selector 'p', text: @usertwo.posts.first.text
+    @usertwo.posts.first.comments.each do |comment|
+      assert_selector 'span', text: comment.user.name
+      assert_selector 'span', text: comment.text
+    end
+  end
```

Comment:
- [ ] Please make sure that you have a Spec file that only tests the `Post` show page :pray:

Reply:
Hi @Lizdev-05  Elizabeth
we did 40 test cases in this project.
inside the user index page we wrote test case assure
    - User index page: visit user_path assert
  - we can see the username of all other users.
  - we can see the profile picture for each user.
  - we can see the number of posts each user has written.
  - When we click on a user, we am redirected to that user's show page.
-  Inside User  show index 
     - we can see the user's profile picture.
  - we can see the user's username.
  - we can see the number of posts the user has written.
  - we can see the user's bio.
  - we can see the user's first 3 posts.
  - we can see a button that lets me view all of a user's posts.
  - When we click a user's post, it redirects me to that post's show page.
  - When we  click to see all posts, it redirects me to the user's post's index page.

Reply:
[my resourec ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/integrations_lesson.md)